lib/Doctrine/ORM/Cache/Region/DefaultRegion.php

```diff
@@ -100,30 +102,28 @@ public function get(CacheKey $key)
      */
     public function getMultiple(CollectionCacheEntry $collection)
     {
-        $keysToRetrieve = array();
-
-        foreach ($collection->identifiers as $index => $key) {
-            $keysToRetrieve[$index] = $this->name . '_' . $key->hash;
-        }
-
-        $items = array_filter(
-            array_map([$this->cache, 'fetch'], $keysToRetrieve),
-            function ($retrieved) {
-                return false !== $retrieved;
+        $result = array();
```

Comment:
Ok, clear. A senior colleague (Java guy) once taught me to use only one
return per function to ease up debugging (only one breakpoint per function
required), since then I adopted it as a daily habit. But it is fine by me,
I regard more as a 'style' thingy ;).

Just pushed and update of the PR.

On 14 April 2015 at 16:15, Guilherme Blanco <notifications@github.com>
wrote:

> In lib/Doctrine/ORM/Cache/Region/DefaultRegion.php
> <https://github.com/doctrine/doctrine2/pull/1382#discussion_r28330003>:
>
> > @@ -100,30 +102,28 @@ public function get(CacheKey $key)
> >       */
> >      public function getMultiple(CollectionCacheEntry $collection)
> >      {
> > -        $keysToRetrieve = array();
> > -
> > -        foreach ($collection->identifiers as $index => $key) {
> > -            $keysToRetrieve[$index] = $this->name . '_' . $key->hash;
> > -        }
> > -
> > -        $items = array_filter(
> > -            array_map([$this->cache, 'fetch'], $keysToRetrieve),
> > -            function ($retrieved) {
> > -                return false !== $retrieved;
> > +        $result = array();
>
> @holtkamp <https://github.com/holtkamp> it's not a convention from
> Doctrine, but we should make it.
> There is clear advantage of applying early returns; you increase code
> readability, reduce the cyclomatic complexity of your algorithm and also
> have fewer bytecode instructions to follow.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/doctrine/doctrine2/pull/1382/files#r28330003>.
>